packages/terra-date-picker/src/DateUtil.jsx

```diff
@@ -142,6 +143,113 @@ class DateUtil {
     return undefined;
   }
 
+  /**
+   * Returns an easily parsable date format string.
+   * @param {string} locale - The locale variant from dateFormatOrder.
+   */
+  static getDateFormat(locale) {
+    let dateVariant;
+
+    if (locale === '0') {
+      dateVariant = 'MM-DD-YYYY';
+    } else if (locale === '1') {
+      dateVariant = 'YYYY-MM-DD';
+    } else {
+      dateVariant = 'DD-MM-YYYY';
+    }
+
+    return dateVariant;
+  }
+
+  /**
+   * Returns an object consisting of placeholder strings and a delimiter,
+   * based on the date format.
+   * @param {string} variant - Variant string obtained from getDateFormat.
+   * @param {string} placeholder - Passed on placeholder prop string.
+   */
+  static getPlaceholderValues(variant, placeholder) {
+    let dateMonth;
+    let dateDay;
+    let dateYear;
+    let dateDelimiter;
+    let ph = placeholder;
+
+    if (!ph) {
+      ph = 'MM-DD-YYYY';
+    }
+
+    if (variant === 'MM-DD-YYYY') {
+      dateMonth = ph.substr(0, 2);
+      dateDay = ph.substr(3, 2);
+      dateYear = ph.substr(6, 4);
+    } else if (variant === 'YYYY-MM-DD') {
+      dateMonth = ph.substr(5, 2);
+      dateDay = ph.substr(8, 2);
+      dateYear = ph.substr(0, 4);
+      dateDelimiter = '-';
+    } else {
+      dateMonth = ph.substr(3, 2);
+      dateDay = ph.substr(0, 2);
+      dateYear = ph.substr(6, 4);
+    }
+
+    if (!dateDelimiter) {
+      dateDelimiter = '/';
```

Comment:
What about when the deliminator should be `.`? https://github.com/cerner/terra-framework/blob/master/packages/terra-date-picker/translations/de.json#L2

Reply:
Oh, good catch! I can add that in. :)